File: leetcode/python/passedProblems/1-two-sum.py
class Solution(object/:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        sortNums=sorted(nums);
        pl = 0;
        pr = len(nums) - 1;
        while (pl < pr):
            sum = sortNums[pl] + sortNums[pr];
            if sum == target:
                break;
            elif sum > target:
                pr -= 1;
            else:
                pl += 1;
        
        if pl == pr:
            return -1,-1;
        else:
            pos1 = nums.index(sortNums[pl]) + 1;
            pos2 = nums.index(sortNums[pr]) + 1;
            if pos1 == pos2:
                pos2 = nums[pos1:].index(sortNums[pr]) + 1 + pos1;
            return min(pos1, pos2), max(pos1, pos2);

        

File: leetcode/cpp/1-two-sum.cpp
/*
 * 找2个下标，使得和为target
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> re;
        vector<pair<int,int> > ns;
        for(int i = 0 ; i < nums.size(); i ++)
            ns.push_back(make_pair(nums[i], i));
        sort(ns.begin(),ns.end());
        int l = 0, r = nums.size()-1;
        while(l < r){
            int sum = ns[l].first + ns[r].first;
            if(sum == target){
                int ia = ns[l].second, ib = ns[r].second;
                re.push_back(ia);
                re.push_back(ib);
                return re;
            }
            else if(sum > target) r --;
            else l ++;
        }
    }
};
